hyper/http20/connection.py

```diff
@@ -544,23 +652,12 @@ def _send_cb(self, frame, tolerate_peer_gone=False):
         and send it on the connection. It does so obeying the connection-level
         flow-control principles of HTTP/2.
         """
-        # Maintain our outgoing flow-control window.
-        if frame.type == DataFrame.type:
-            # If we don't have room in the flow control window, we need to look
-            # for a Window Update frame.
-            while self._out_flow_control_window < len(frame.data):
-                self._recv_cb()
-
-            self._out_flow_control_window -= len(frame.data)
-
         data = frame.serialize()
-
         max_frame_size = self._settings[SettingsFrame.SETTINGS_MAX_FRAME_SIZE]
```

Comment:
So anything that comes off of settings can be changed under our feet. I suspect this means we'll need to have an RWLock around settings access that ensures that while we're using a settings value it doesn't change.